src/helpers.js

```diff
@@ -195,6 +204,230 @@ module.exports = ({ types: t }) => {
+              )
+            ),
+            t.blockStatement([
+              t.returnStatement(t.callExpression(t.identifier(REQUIRE), [depsIdentifier])),
+            ])
+          )
+        );
+      }
+      // Convert array-like objects to real arrays
+      // amdDeps = [].slice.call(amdDeps);
+      blockStatements.push(
+        t.expressionStatement(
+          t.assignmentExpression(
+            '=',
+            depsIdentifier,
+            t.callExpression(
+              t.memberExpression(
+                t.memberExpression(t.arrayExpression(), t.identifier('slice')),
+                t.identifier('call')
+              ),
+              [depsIdentifier]
+            )
+          )
+        )
+      );
+    }
+
+    // If we don't know that the factory is a function, then we need to check the type
+    // at runtime.
+    if (!isFunctionExpression(factory)) {
+      // define:
+      // if (typeof maybeFunction !== 'function') {
+      //   var amdFactoryResult = maybeFunction;
+      //   maybeFunction = function () {return amFactorydResult};
+      // }
+      // require:
+      // if (typeof maybeFunction !== 'function') {
+      //   maybeFunction = function () {};
+      // }
+      const resultIdentifier = getUniqueIdentifier(path.scope, AMD_FACTORY_RESULT);
+      blockStatements.push(
+        t.ifStatement(
+          t.binaryExpression(
+            '!==',
+            t.unaryExpression('typeof', factoryIdentifier),
+            t.stringLiteral('function')
+          ),
+          t.blockStatement([
+            ...(isDefineCall
+              ? [
+                  t.variableDeclaration('var', [
+                    t.variableDeclarator(resultIdentifier, factoryIdentifier),
+                  ]),
+                ]
+              : []),
+            t.expressionStatement(
+              t.assignmentExpression(
+                '=',
+                factoryIdentifier,
+                t.functionExpression(
+                  null,
+                  [],
+                  t.blockStatement([...(isDefineCall ? [t.returnStatement(resultIdentifier)] : [])])
+                )
+              )
+            ),
+          ])
+        )
+      );
+    }
+    // Invoke the factory function.
+    const factoryInvocation = t.callExpression(
+      t.memberExpression(factoryIdentifier, t.identifier('apply')),
+      [
+        /*
+         * The 'this' argument for the factory function.
+         * 'void 0'
+         */
+        t.unaryExpression('void', t.numericLiteral(0)),
+
+        /*
+         * The arguments to the factory function as an array.
+         * 'deps.map(<amdFactoryArgsMapper>)''
+         */
+        t.callExpression(t.memberExpression(depsIdentifier, t.identifier('map')), [
+          getAmdFactoryArgsMapper(path, opts),
+        ]),
+      ]
+    );
+    if (isDefineCall) {
+      const resultCheckIdentifier = getUniqueIdentifier(path.scope, AMD_DEFINE_RESULT);
+      blockStatements.push.apply(
```

Comment:
Not exactly the same because `createModuleExportsResultCheck` returns an array of block statements.  You're suggested change will push the array object onto `blockStatements` rather than adding the individual elements to the `blockStatements` array.  We could use `blockStatements.concat` instead, but that requires a re-assignment and I don't see a reason to prefer one approach over the other.

Reply:
Ah I see, let's use the spread operator to make it clearer what's happening then. (The source is transpiled before publish time so this is fine for backcompat.) e.g. `blockStatements.push(...arr)`

Reply:
Ah, that wrapper.  I thought we were talking about the outer wrapper.  So the reason for using apply here is so that the function arguments can be passed as an array (the result of calling `amdDeps.map`).

Reply:
Ahh got it, thanks for the explanation